### PR TITLE
feature of signup

### PR DESCRIPTION
**Pull request by brunomichetti**
- Communication with backend and handle signup error.
- Feedback for: email already registered, error in password, missing fields, different password and password confirm, error in server.
- Standard applied to all functions definitions.
- Modularization of form input error to a new component

TODO:
- style for react select
- confirm account with email
Screen Shot 2020-06-29 at 13 08 45
Screen Shot 2020-06-29 at 13 09 06
Screen Shot 2020-06-29 at 13 09 53
Screen Shot 2020-06-29 at 13 10 05

### CODE REVIEW

File: src/actions/user.actions.js [hunk 1]
```diff
@@ -52,7 +61,34 @@ const logout = () => {
...
+  const success = (user) => {
+    return { type: userConstants.SIGNUP_SUCCESS, user };
+  };
+  const failure = (error) => {
+    return { type: userConstants.SIGNUP_FAILURE, error };
+  };
+  return async (dispatch) => {
+    dispatch(request({ email }));
+    try {
+      const response = await userService.signup(
```
Comment (manema) [inline src/actions/user.actions.js]:
```suggestion
      const { data: { user }} = await userService.signup(
        name,
        email,
        password1,
        password2,
        gender
      );
      success_to_homepage(user, dispatch, success);
```

File: src/helpers/error.handler.js [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
     return loginErrorConstants.LOGIN_ERROR_CREDENTIALS;
   } else {
-    return loginErrorConstants.LOGIN_SERVER_ERROR;
+    return loginErrorConstants.SERVER_ERROR;
   }
 };
 
-export default handleLoginError;
+export const handleSignupError = (error) => {
+  debugger;
```
Comment (manema) [inline src/helpers/error.handler.js]: ☝️

File: src/helpers/error.handler.js [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
   if (errorResponse.status === 400 && errorResponse.data.non_field_errors) {
     return loginErrorConstants.LOGIN_ERROR_CREDENTIALS;
   } else {
-    return loginErrorConstants.LOGIN_SERVER_ERROR;
+    return loginErrorConstants.SERVER_ERROR;
   }
 };
 
-export default handleLoginError;
+export const handleSignupError = (error) => {
```
Comment (manema) [inline src/helpers/error.handler.js]:
```suggestion
export const handleSignupError = ({ response: { data } }) => {
  if (data['email']) return data['email']
  else if (data['password1']) return data.password1[0]
  return loginErrorConstants.SERVER_ERROR;
};
```

File: src/components/users/UserFormInputError.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
Comment (manema) [inline src/components/users/UserFormInputError.js]: In general we add the error to the input component itself.

**Review by manema — APPROVED** [review]

File: src/actions/user.actions.js [hunk 5]
```diff
@@ -2,45 +2,53 @@ import { userConstants } from '../constants/user.constants';
...
 import { alertActions } from '../actions/alert.actions';
-import handleLoginError from '../helpers/error.handler';
+import { handleLoginError, handleSignupError } from '../helpers/error.handler';
 import { homePageLink, loginPageLink } from '../constants/link.constants';
 
+const request = (user) => {
+  return { type: userConstants.USER_REQUEST, user };
+};
+
+const success_to_homepage = (user, dispatch, success) => {
```
Comment (tamarasuarezrod) [inline src/actions/user.actions.js]: use camel case

File: src/actions/user.actions.js [hunk 6]
```diff
@@ -2,45 +2,53 @@ import { userConstants } from '../constants/user.constants';
...
+  localStorage.setItem('user', JSON.stringify(user));
+  dispatch(success(user));
+  dispatch(alertActions.success());
+  history.push(homePageLink);
+};
+
+const error_from_server = (errorMsg, dispatch, failure) => {
```
Comment (tamarasuarezrod) [inline src/actions/user.actions.js]: same here

**Review by tamarasuarezrod — APPROVED** [review]

**Pull request merged**